config/config.go

```diff
@@ -3,48 +3,209 @@ package config
+	SupportedFiles []string       `yaml:"supported_files"`
+	ModKey         string         `yaml:"mod_key"`
+}
+
+var yamlData = AllConfig{
+	Instance: InstanceConfig{
+		Domain:      "",
+		Port:        3000,
+		Protocol:    "https",
+		Name:        "",
+		Summary:     "",
+		Salt:        "",
+		MaxFilesize: 8,
+	},
+	Email: EmailConfig{
+		Address:  "",
+		Password: "",
+		Host:     "localhost",
+		Port:     587,
+		Notify:   []string{},
+	},
+	Database: DatabaseConfig{
+		Host:     "localhost",
+		Port:     5432,
+		Name:     "postgres",
+		User:     "postgres",
+		Password: "",
+	},
+	TorProxy:    "127.0.0.1:9050",
+	CookieKey:   "",
+	AuthReq:     []string{"captcha", "Email", "passphrase"},
+	PostPerPage: 10,
+	SupportedFiles: []string{
+		"image/gif", "image/jpeg", "image/png", "image/webp", "image/apng",
+		"video/mp4", "video/ogg", "video/webm",
+		"audio/mpeg", "audio/ogg", "audio/wav", "audio/wave", "audio/x-wav"},
+	ModKey: "",
+}
+
+var Port string
+var TP string
+var Domain string
+var InstanceName string
+var InstanceSummary string
+var Salt string
+var MaxFilesize int64
+
+var SiteEmail string //contact@fchan.xyz
+var SiteEmailPassword string
+var SiteEmailServer string //mail.fchan.xyz
+var SiteEmailPort string   //587
+var SiteEmailNotifyTo string
+
+var DBHost string
+var DBPort int
+var DBUser string
+var DBPassword string
+var DBName string
+
+var TorProxy string //127.0.0.1:9050
+var CookieKey string
+var AuthReq []string
+var PostCountPerPage int
+var SupportedFiles []string
+var Key string
+
 var ActivityStreams = "application/ld+json; profile=\"https://www.w3.org/ns/activitystreams\""
-var AuthReq = []string{"captcha", "email", "passphrase"}
-var PostCountPerPage = 10
-var SupportedFiles = []string{"image/gif", "image/jpeg", "image/png", "image/webp", "image/apng", "video/mp4", "video/ogg", "video/webm", "audio/mpeg", "audio/ogg", "audio/wav", "audio/wave", "audio/x-wav"}
 var Log = log.New(os.Stdout, "", log.Ltime)
 var MediaHashs = make(map[string]string)
-var Key = GetConfigValue("modkey", "")
 var Themes []string
 var DB *sql.DB
 
-// TODO Change this to some other config format like YAML
-// to save into a struct and only read once
-func GetConfigValue(value string, ifnone string) string {
+func LoadConfig() {
+
+	file, err := os.ReadFile("config/config-init.yaml")
+
+	if os.IsNotExist(err) {
+		Log.Println("Failed to read config/config-init.yaml, creating new")
+		if _, err := os.Stat("config/config-init"); err == nil {
+			migrateOldConfig()
+		}
+
+		data, err2 := yaml.Marshal(&yamlData)
+		if err2 != nil {
+			Log.Fatalln(err2)
+		}
+
+		err2 = os.WriteFile("config/config-init.yaml", data, 0640)
+		if err2 != nil {
+			Log.Fatalln(err2)
+		}
+
+		Log.Println("New config written to config/config-init.yaml")
+	} else if err != nil {
+		Log.Fatalln(err)
+	}
+
+	err = yaml.Unmarshal(file, &yamlData)
+	if err != nil {
+		Log.Fatalln(err)
+	}
+
+	// Instance configuration
+	TP = yamlData.Instance.Protocol + "://"
```

Comment:
If you have to generate the YAML config file from Docker, why not just introduce a light wrapper that automatically chooses between environment variables and the configuration file? Something like this:

```go
func envOrOther(env string, other string) string {
  if e := os.Getenv(env); e != "" {
    return e
  }
  return other
}
```